test/suites/integration/deregister_targets_test.go

```diff
@@ -78,21 +75,42 @@ var _ = Describe("Deregister Targets", func() {
 		)
 	})
 
-	It("Kubernetes Service deletion deregisters targets", func() {
+	It("Kubernetes Service deletion triggers targets de-registering", func() {
+		Skip("Currently controller have a bug, service deletion do NOT triggers targets de-registering, need to further investigate the root cause")
 		testFramework.ExpectDeleted(ctx, service)
```

Comment:
After fixed the obvious mistake for this test case (by adding the `.Should(Succeed())`), I found currently controller don't have this behaviour "Kubernetes Service deletion triggers targets de-registering"

I also did manual test several times and could confirm that is a bug.

I think we should temprarily skip() this test case and investigate the root cause in another issue:
https://github.com/aws/aws-application-networking-k8s/issues/331

Reply:
I dont think we should skip test because of bug, we need to address bug. Let it fail until fixed.

Reply:
But before we fix it it always leave a failed test cases? and other developer could be confuse on whether their code change break something but actually not? Also if we leave it not skip(), in the Eventually() block it found the targets still existing over and over again, and finally failed on Eventually() block timeout, it can take 3 min, I think this 3 min wait time is meaningless

Reply:
Then we should fix it as first priority, disabling tests because of bugs is not a good idea.

Reply:
I think we could use `Fail()`

Reply:
Looking again into this problem, why dont we focus on fixing this bug and come back to this PR after?

Reply:
I think bug fixing and e2e test changes should come together, not in this PR